src/main/java/org/javarosa/core/model/instance/VirtualDataInstance.java

```diff
@@ -11,69 +11,23 @@
 /**
  * Data instance that is created during execution based on user input of other live data
  */
-public class VirtualDataInstance extends DataInstance {
-    private TreeElement root = new TreeElement();
-    private InstanceBase base;
+public class VirtualDataInstance extends ExternalDataInstance {
```

Comment:
This is now exactly same as `ExternalDataInstance` . Should we now remove the concept of `VirtualDataInstance` completely ? Not sure if it might still be potentially useful to keep this distinction.